gobblin-cluster/src/main/java/gobblin/cluster/GobblinHelixTask.java

```diff
@@ -113,10 +111,15 @@ public TaskResult run() {
       Path workUnitFilePath =
           new Path(this.taskConfig.getConfigMap().get(GobblinClusterConfigurationKeys.WORK_UNIT_FILE_PATH));
 
-      WorkUnit workUnit =
-          workUnitFilePath.getName().endsWith(AbstractJobLauncher.MULTI_WORK_UNIT_FILE_EXTENSION) ? MultiWorkUnit
-              .createEmpty() : WorkUnit.createEmpty();
-      SerializationUtils.deserializeState(this.fs, workUnitFilePath, workUnit);
+      String fileName = workUnitFilePath.getName();
```

Comment:
There are still a lot of references to file system style storage (e.g. file, directory, etc.) that might be confusing in the future. How hard would it be to refactor into the correct terms for a generic store?